app/src/main/java/org/openobservatory/ooniprobe/common/MKCollectorResubmitTask.java

```diff
@@ -0,0 +1,98 @@
+package org.openobservatory.ooniprobe.common;
+
+import android.content.Context;
+import android.util.Log;
+import android.view.WindowManager;
+
+import com.raizlabs.android.dbflow.sql.language.OperatorGroup;
+import com.raizlabs.android.dbflow.sql.language.SQLOperator;
+import com.raizlabs.android.dbflow.sql.language.SQLite;
+
+import org.apache.commons.io.FileUtils;
+import org.openobservatory.ooniprobe.R;
+import org.openobservatory.ooniprobe.model.database.Measurement;
+import org.openobservatory.ooniprobe.model.database.Measurement_Table;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+
+import androidx.appcompat.app.AppCompatActivity;
+import io.ooni.mk.MKCollectorResubmitResults;
+import io.ooni.mk.MKCollectorResubmitSettings;
+import localhost.toolkit.os.NetworkProgressAsyncTask;
+
+public class MKCollectorResubmitTask<A extends AppCompatActivity> extends NetworkProgressAsyncTask<A, Integer, Void> {
+	/**
+	 * Use this class to resubmit a measurement, use result_id and measurement_id to filter list of value
+	 * {@code new MKCollectorResubmitTask(activity).execute(@Nullable result_id, @Nullable measurement_id);}
+	 *
+	 * @param activity from which this task are executed
+	 */
+	public MKCollectorResubmitTask(A activity) {
+		super(activity, true, false);
+	}
+
+	private static void perform(Context c, Measurement m) throws IOException {
+		String input = FileUtils.readFileToString(Measurement.getEntryFile(c, m.id, m.test_name), StandardCharsets.UTF_8);
+		MKCollectorResubmitSettings settings = new MKCollectorResubmitSettings();
+		settings.setTimeout(14);
+		settings.setCABundlePath(c.getCacheDir() + "/" + Application.CA_BUNDLE);
+		settings.setSerializedMeasurement(input);
+		MKCollectorResubmitResults results = settings.perform();
+		if (results.isGood()) {
+			String output = results.getUpdatedSerializedMeasurement();
+			FileUtils.writeStringToFile(Measurement.getEntryFile(c, m.id, m.test_name), output, StandardCharsets.UTF_8);
+			m.report_id = results.getUpdatedReportID();
+			m.is_uploaded = true;
+			m.is_upload_failed = false;
+			m.save();
+		} else {
+			Log.w(MKCollectorResubmitSettings.class.getSimpleName(), results.getLogs());
+			// TODO decide what to do with logs (append on log file?)
+		}
+	}
+
+	@Override protected void onPreExecute() {
+		super.onPreExecute();
+		A activity = getActivity();
+		if (activity != null)
+			activity.getWindow().addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON);
+	}
+
+	@Override protected Void doInBackground(Integer... params) {
```

Comment:
A docstring should be added explaining that this function takes two positional arguments which are one the `measurement_id` and the other is the `result_id`. If it has both arguments set to `null` it will then upload all measurements.

Reply:
There is on constructor because you will not call this method directly

Reply:
https://github.com/ooni/probe-android/pull/217#discussion_r277808743